### ISSUE TITLE
Type compatibility issues with material-ui 4.9.0

### ISSUE THREAD

**Issue by mvestergaard** [mui-org/material-ui-pickers#1465]
Due to the changes made in https://github.com/mui-org/material-ui/pull/19320 a type error occurs in pickers

```
node_modules/@material-ui/pickers/views/Clock/Clock.d.ts:4:48 - error TS2344: Type '(theme: Theme) => Record<"container" | "clock" | "squareMask" | "pin", CSSProperties | (() => CSSProperties)>' does not satisfy the constraint 'string | Record<any, CSSProperties | CreateCSSProperties<any> | ((props: any) => CreateCSSProperties<any>)> | StyleRulesCallback<...>'.
  Type '(theme: Theme) => Record<"container" | "clock" | "squareMask" | "pin", CSSProperties | (() => CSSProperties)>' is not assignable to type 'StyleRulesCallback<any, any, any>'.
    Type 'Record<"container" | "clock" | "squareMask" | "pin", CSSProperties | (() => CSSProperties)>' is not assignable to type 'Record<any, CSSProperties | CreateCSSProperties<any> | ((props: any) => CreateCSSProperties<any>)>'.
      Property 'container' is incompatible with index signature.
        Type 'CSSProperties | (() => CSSProperties)' is not assignable to type 'CSSProperties | CreateCSSProperties<any> | ((props: any) => CreateCSSProperties<any>)'.
          Type '() => CSSProperties' is not assignable to type 'CSSProperties | CreateCSSProperties<any> | ((props: any) => CreateCSSProperties<any>)'.
            Type '() => CSSProperties' is not assignable to type '(props: any) => CreateCSSProperties<any>'.

export interface ClockProps extends WithStyles<typeof styles> {
                                                 ~~~~~~~~~~~~~
```

## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers |3.2.9         |
| material-ui          |4.9.0         |
| TypeScript           |3.7.5         |

**Comment by OriginLive** [comment]
Hi, any eta on when this will be resolved?

**Comment by dmtrKovalenko** [comment]
Released in 2.3.10
